File: chatwoot/python/src/UpdateAutomationRuleInAccountExample.py
import json
from datetime import date, datetime
from pprint import pprint

from chatwoot_client import ApiClient, ApiException, Configuration, api, models

configuration = Configuration(
    api_key={"userApiKey": "USER_API_KEY"},
)

with ApiClient(configuration) as api_client:
    automation_rule_create_update_payload = models.AutomationRuleCreateUpdatePayload(
        name="Add label on message create event",
        description="Add label support and sales on message create event if incoming message content contains text help",
        event_name="message_created",
        active=None,
    )

    try:
        response = api.AutomationRuleApi(api_client).update_automation_rule_in_account(
            account_id=None,
            id=None,
            automation_rule_create_update_payload=automation_rule_create_update_payload,
        )

        pprint(response)
    except ApiException as e:
        print("Exception when calling AutomationRuleApi#update_automation_rule_in_account: %s\n" % e)

File: chatwoot/ruby/src/UpdateAutomationRuleInAccountExample.rb
require "json"
require "chatwoot_client"

ChatwootClient.configure do |config|
    config.api_key["userApiKey"] = "USER_API_KEY"
end

automation_rule_create_update_payload = ChatwootClient::AutomationRuleCreateUpdatePayload.new
automation_rule_create_update_payload.name = "Add label on message create event"
automation_rule_create_update_payload.description = "Add label support and sales on message create event if incoming message content contains text help"
automation_rule_create_update_payload.event_name = "message_created"
automation_rule_create_update_payload.active = nil

begin
    response = ChatwootClient::AutomationRuleApi.new.update_automation_rule_in_account(
        nil, # account_id
        nil, # id
        automation_rule_create_update_payload, # data
    )

    p response
rescue ChatwootClient::ApiError => e
    puts "Exception when calling AutomationRuleApi#update_automation_rule_in_account: #{e}"
end
